go/vt/vtgate/planbuilder/operators/route_planning.go

```diff
@@ -548,9 +548,11 @@ func unwrapDerivedTables(ctx *plancontext.PlanningContext, exp sqlparser.Expr) s
 		}
 
 		exp = semantics.RewriteDerivedTableExpression(exp, tbl)
-		exp = getColName(exp)
-		if exp == nil {
+		switch col := getColName(exp); col {
+		case nil:
 			return nil
+		default:
+			exp = col
```

Comment:
Is this more clear with the following logic vs. the switch?

```go
if col := getColName(exp); col == nil {
   return nil
}
exp = col
```

Reply:
my concern with this one specifically is idk if this is taking advantage of a scoping thing that could get fixed along with the "loop variable scope" issue (what i am saying is that regardless of how go actually treats this, i think `col` _shouldn't_ be in scope outside the if

so you'd need instead to do:

```go
if col := getColName(exp); col != nil {
    exp = col
} else {
    return nil
}
```

which isn't really more concise (but i could see an argument for still being more _clear_) than the switch version

Reply:
Doh, you're right. It would have to be something like  this to avoid the `else`:

```go
col := getColName(exp)
if col == nil {
   return nil
}
exp = col
```

Again, no super strong opinion here.